src/test/java/com/puppycrawl/tools/checkstyle/api/FullIdentTest.java

```diff
@@ -48,6 +48,40 @@ public void testNonValidCoordinatesWithZero() {
         Assert.assertEquals("Invalid full indent", "MyTest.MyTestik[15x14]", fullIdent.toString());
     }
 
+    @Test
+    public void testWithArrayCreateFullIdentWithArrayDeclare() {
+        final DetailAST ast = new DetailAST();
+        ast.setType(TokenTypes.ARRAY_DECLARATOR);
+        ast.setColumnNo(18);
+        ast.setLineNo(15);
+        ast.setText("[");
```

Comment:
We try to avoid such tests as much as possible, please make such case in non compilable sources if you need.
All test coverage should be on real Java files , not a fake trees.

Reply:
That was from Pull #5073 
It has been rewritten with input file parsing by TreeWalker.

Reply:
@kazachka ,

this is not good reason :), we do not need such hacks. We do not allow in UT use pure-UT 
model to test modules as it pure-UT are doe snot test real testcases and produce a lot of test code that actually have no value as such trees are not possible to make by java code (dead code).

please make a real java file and let it be processed as in all other test methods.

> please make such case in non compilable sources if you need.

it is really last resort.
As our main limitation is that we do not work on non-compiled sources.
So there should be very good reason for such file too appear.

Reply:
looks like resolved.